test/miscellaneous/odp_dyn_workers.c

```diff
@@ -767,59 +932,47 @@ static int send_command(int socket, uint8_t cmd)
 	return data;
 }
 
-static void dump_summary(int socket, pid_t pid)
+static odp_bool_t recv_summary(int socket, summary_t *summary)
 {
-	summary_t stats;
-	const ssize_t size = sizeof(stats), ret = TEMP_FAILURE_RETRY(recv(socket, &stats, size,
-									  0));
+	const ssize_t size = sizeof(*summary),
+	ret = TEMP_FAILURE_RETRY(recv(socket, summary, size, 0));
 
-	if (ret < size)
-		return;
+	return ret == size;
+}
 
+static void dump_summary(pid_t pid, const summary_t *summary)
+{
 	printf("\nremoved worker summary:\n"
 	       "    ODP process ID: %d\n"
 	       "    thread ID:      %" PRIu64 "\n"
 	       "    events handled: %" PRIu64 "\n"
 	       "    enqueue errors: %" PRIu64 "\n"
-	       "    runtime:        %" PRIu64 " (ns)\n\n", pid, stats.thread_id, stats.num_handled,
-	       stats.enq_errs, stats.runtime);
+	       "    runtime:        %" PRIu64 " (ns)\n\n", pid, summary->thread_id,
+	       summary->num_handled, summary->enq_errs, summary->runtime);
 }
 
-static void run_global(global_config_t *config)
+static odp_bool_t check_summary(const summary_t *summary)
 {
-	char *input;
-	char cmdstr[MAX_CMD_LEN + 1U];
-	size_t size;
+	return summary->num_handled > 0U && summary->enq_errs == 0U && summary->runtime > 0U;
```

Comment:
It could be helpful to print some debug info in case of failures. E.g. while testing the PR it took some figuring out why 
```
$ ./odp_dyn_workers -c 0xf0,0xf00 -p a0,d1000000,r0,d1000000,a1,d1000000,r1
```
kept failing and output was only printed from one process.

Reply:
Why did it fail?

Reply:
I didn't dig up the source, but after increasing the delay to 10 000 000 ns the command passes.

Reply:
OK, probably then due to no handled events (if the delay is too short, worker main thread doesn't have time to execute the schedule loop). Added additional logging in v3.